File: src/resolvers/mypix.py
# -*- coding: utf-8 -*-

import re

class mypix:
	def __str__(self):
		return "MyPix"
	
	regexStr = "^http://www\\.mypix\\.jp/app.php/picture/(\\d+)/?(?:\\?.*)?$"
	regex = re.compile(regexStr, re.IGNORECASE)
	
	def getFullSize(self, match):
		return "http://www.mypix.jp/app.php/picture/%s/860x0.jpg" % match.group(1)
	
	def getLargeSize(self, match):
		return "http://www.mypix.jp/app.php/picture/%s/860x0.jpg" % match.group(1)
	
	def getThumbnail(self, match):
		return "http://www.mypix.jp/app.php/picture/%s/thumb.jpg" % match.group(1)

File: ImgAzyobuziNet.Core/Resolvers/MyPix.cs
using System.Text.RegularExpressions;
using System.Threading.Tasks;
using ImgAzyobuziNet.TestFramework;
using Shouldly;

namespace ImgAzyobuziNet.Core.Resolvers
{
    public class MyPixProvider : PatternProviderBase<MyPixResolver>
    {
        public override string ServiceId => "MyPix";

        public override string ServiceName => "MyPix";

        public override string Pattern => @"^http://www\.mypix\.jp/app.php/picture/(\d+)(?:/(?:in/?)?)?(?:[\?#]|$)";

        #region Tests

        [TestMethod(TestCategory.Static)]
        private void RegexTest()
        {
            var match = this.GetRegex().Match("http://www.mypix.jp/app.php/picture/64303/in");
            match.Success.ShouldBeTrue();
            match.Groups[1].Value.ShouldBe("64303");
        }

        #endregion
    }

    public class MyPixResolver : IResolver
    {
        public ValueTask<ImageInfo[]> GetImages(Match match)
        {
            var id = match.Groups[1].Value;
            var fullUri = $"http://www.mypix.jp/app.php/picture/{id}/860x0.jpg";
            return new ValueTask<ImageInfo[]>(new[]
            {
                new ImageInfo(fullUri, fullUri, $"http://www.mypix.jp/app.php/picture/{id}/thumb.jpg")
            });
        }
    }
}
